### DISCUSSION TITLE
deck.pickObject returns invalid info when using MVTLayer in MapboxLayer

### DISCUSSION THREAD

**Discussion by fifzteen** [visgl/deck.gl#5671] [Q&A] [answered]
Look my example. (input your mapbox accessToken.)
https://codepen.io/fifzteen/pen/JjEZvEL

At first, the center of view is [139.8, 35.7].

![139_8-35_7](https://user-images.githubusercontent.com/25410805/114961890-abdacb00-9ea4-11eb-97c8-82659a6f0e3d.png)

Click the polygon inside the red rectangle, `pickObject` returns valid info. (logged to console below)
```
(clicked lat) (clicked lng) {"orgGILvl": "2500", "ftCode": 3102, "lvOrder": 0,"layerName": "building"}
```

Next, scroll left, to [139.7925, 35.7].

![139_7925-35_7](https://user-images.githubusercontent.com/25410805/114961896-ae3d2500-9ea4-11eb-9e76-479e8874dff6.png)

Click the polygon inside the red circle.
expect log is below.

```
(clicked lat) (clicked lng) {"orgGILvl": "2500", "ftCode": 5000, "layerName": "waterarea"}
```

actually, `pickObject` returns invalid info. logged to console below.

```
(clicked lat) (clicked lng) {"orgGILvl": "2500", "ftCode": 3102, "lvOrder": 0,"layerName": "building"}
or
(clicked lat) (clicked lng) null
```

What causes this?
It looks the off-scrren buffer is not updated when viewState changing?

**Comment by padawannn** [comment, marked as answer]
You need to update the deckGl viewstate whenever the map moves.  Try adding a code like this to your example:
```
map.on('moveend', function() {
  const center = map.getCenter();
  const zoom = map.getZoom();

  deckGl.setProps({viewState:{
    longitude: center.lng,
    latitude: center.lat,
    zoom
  }});
})
```

**Reply by fifzteen** [reply]
@padawannn 
Thank you for your reply.
I add updating viewState callback you introduced, then pickObject returns the correct info!
